source/server/active_tcp_listener.h

```diff
@@ -81,6 +85,9 @@ class ActiveTcpListener final : public Network::TcpListenerCallbacks,
   // The number of connections currently active on this listener. This is typically used for
   // connection balancing across per-handler listeners.
   std::atomic<uint64_t> num_listener_connections_{};
+
+  Network::ConnectionBalancer& connection_balancer_;
+  Network::Address::InstanceConstSharedPtr address_;
```

Comment:
This listener must know it's address because it has a socket, right? It seems like this shouldn't be necessary?

Reply:
I have think about another two ways:
* We can get the address from the accepted socket, but it won't work with the case of the connection doesn't use original destination.
* The socket object is held by the socket listener, then we can add an interface to the listener to return the socket address.
   https://github.com/envoyproxy/envoy/blob/17d7d7908507888617824d73bc5904a3d79d1d17/envoy/network/listener.h#L398

The second way works also. But let me know if the second way you think better.

Reply:
I add a comment here https://github.com/envoyproxy/envoy/pull/21774/files#diff-55b19f848d75e4c86be99ac5c87457e24a960e71913de4b60153ea5e47c4d32cR90

But let me know if you prefer another way.